tests/core/vm/test_vm.py

```diff
@@ -46,3 +53,28 @@ def test_import_block(chain_without_block_validation):  # noqa: F811
     parent_vm = chain.get_chain_at_block_parent(vm.block).get_vm()
     block = parent_vm.import_block(vm.block)
     assert block.transactions == [tx]
+
+
+def test_sharding_transaction(shard_chain_without_block_validation):  # noqa: F811
+    # Target contract, i.e. funded_address, is a simple ether transfer contract which transfer
+    # specified amount of ether to recipient.
+    # TODO: update contract_deployment_code
+    contract_deployment_code = b''
+    contract_addr = generate_create2_contract_address(b'', contract_deployment_code)
+
+    chain = shard_chain_without_block_validation(contract_addr)
+    deploy_tx = new_sharding_transaction(contract_addr, b'', 0, b'', b'', contract_deployment_code)
+
+    vm = chain.get_vm()
+    computation = vm.apply_transaction(deploy_tx)
+    assert not computation.is_error
+
+    # Transfer ether to recipient
+    recipient = decode_hex('0xa94f5374fce5edbc8e2a8697c15331677e6ebf0c')
+    amount = 100
+    tx_initiator = chain.funded_address
+    transfer_tx = new_sharding_transaction(tx_initiator, recipient, amount, b'', b'', b'')
+    computation = vm.execute_transaction(transfer_tx)
+    assert not computation.is_error
+    with vm.state.state_db(read_only=True) as state_db:
```

Comment:
Just had the thought that we could provide access to a read-only statedb without the context manager as part of the `state` object, and have the context manager API be only necessary for when write operations are necessary.

Reply:
That makes sense. Just to make sure, context manager's job here is to (1) double check no state were written in read_only mode and (2) decommission the state to prevent further modification to db, correct?
Does it involve managing(opening/closing/cleaning) the actual data base file?